tests/src/test_settings_util.cpp

```diff
@@ -52,8 +71,16 @@ TEST(test_settings_util, parse) {
   EXPECT_STREQ("value", value);
   EXPECT_STREQ("type", type);
 
+  /* Check that old style double NULL at end is not accepted */
   char sect_name_val_enumtype[27] = {'s', 'e', 'c', 't', '\0', 'n', 'a', 'm', 'e', '\0', 'v', 'a', 'l', 'u', 'e', '\0', 'e', 'n', 'u', 'm', ',', 't', 'y', 'p', 'e', '\0', '\0'};
-  EXPECT_EQ(SETTINGS_TOKENS_TYPE_CUSTOM, settings_parse(sect_name_val_enumtype, sizeof(sect_name_val_enumtype), &section, &name, &value, &type));
+  EXPECT_EQ(SETTINGS_TOKENS_INVALID, settings_parse(sect_name_val_enumtype, sizeof(sect_name_val_enumtype), &section, &name, &value, &type));
```

Comment:
If this is integrated into the new console, we would want to ensure backwards compatibility with a piksi running older firmware. Does this test prevent that?

Reply:
This seems to be a relevant concern. Old `settings_format_setting` is used in
- settings_register_callback
- settings_read_callback
- settings_read_by_index_callback
I'll fix this.